chapter_optimization/adagrad.md

```diff
@@ -139,11 +147,26 @@ def adagrad(params, states, hyperparams):
         p.grad.data.zero_()
 ```
 
+```{.python .input}
+#@tab tensorflow
+def init_adagrad_states(feature_dim):
+    s_w = tf.Variable(d2l.zeros((feature_dim, 1)))
+    s_b = tf.Variable(d2l.zeros(1))
+    return (s_w, s_b)
+
+def adagrad(params, grads, states, hyperparams):
+    eps = 1e-6
+    for p, s, g in zip(params, states, grads):
+        s[:].assign(s + tf.math.square(g))
+        p[:].assign(p - hyperparams['lr'] * g / tf.math.sqrt(s + eps))
+```
+
 Compared to the experiment in :numref:`sec_minibatch_sgd` we use a
 larger learning rate to train the model.
 
 ```{.python .input}
 #@tab all
+# todo fix adagrad
```

Comment:
Is this fixed?

Reply:
Yes, it's fixed. Sorry, actually I forget to remove it.